### PR TITLE
DataONE Harvester & Normalizer

### PR DESCRIPTION
**Pull request by caseyrollins**
Depends on #207

### CODE REVIEW

**Comment by caseyrollins** [comment]
@chrisseto 🎲

File: providers/org/dataone/normalizer.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    )
+    links = Map(
+        Delegate(ThroughLinks),
+        Maybe(XPath(ctx, "str[@name='dataUrl']"), 'str')['#text']
+    )
+
+    class Extra:
+        datasource = Maybe(XPath(ctx, "str[@name='datasource']"), 'str')['#text']
+        dateModified = Maybe(XPath(ctx, "date[@name='dateModified']"), 'date')['#text']
+        datePublished = Maybe(XPath(ctx, "date[@name='datePublished']"), 'date')['#text']
```
Comment (chrisseto) [inline providers/org/dataone/normalizer.py]: Date published exists on `CreativeWork`

File: providers/org/dataone/normalizer.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        Maybe(XPath(ctx, "arr[@name='keywords']"), 'arr').str
+    )
+    links = Map(
+        Delegate(ThroughLinks),
+        Maybe(XPath(ctx, "str[@name='dataUrl']"), 'str')['#text']
+    )
+
+    class Extra:
+        datasource = Maybe(XPath(ctx, "str[@name='datasource']"), 'str')['#text']
+        dateModified = Maybe(XPath(ctx, "date[@name='dateModified']"), 'date')['#text']
```
Comment (chrisseto) [inline providers/org/dataone/normalizer.py]: @erinspace was maybe going to add `date_modified` to creative work?

File: providers/org/dataone/normalizer.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+class Contributor(Parser):
+    order_cited = ctx('index')
+    cited_name = ctx
+    person = Delegate(Person, ctx)
+
+
+class CreativeWork(Parser):
+    title = Maybe(XPath(ctx, "str[@name='title']"), 'str')['#text']
+    description = Maybe(XPath(ctx, "str[@name='abstract']"), 'str')['#text']
+    published = Maybe(XPath(ctx, "date[@name='datePublished']"), 'date')['#text']
```
Comment (chrisseto) [inline providers/org/dataone/normalizer.py]: `ParseDate` pls

File: providers/org/dataone/normalizer.py (hunk 1 above)
Comment (caseyrollins) [inline providers/org/dataone/normalizer.py]: DataONE provides both a ```datePublished``` and ```pubDate```, which, in all the cases I saw, are exactly the same thing. I'm using ```pubDate``` as the ```published``` field on the ```CreativeWork``` ,and storing the secondary date in ```Extra```.

**Comment by caseyrollins** [comment]
@chrisseto updated 🍀

**Pull request merged**